### ISSUE TITLE
Failure in mixer init

### ISSUE THREAD

**Issue by wizzardx** [Vladar4/nimgame2#13]
Hi there.

On my system, with the first tut, I get this error:

CRITICAL: Can't initialize SDL_mixer: OGG support not available

I'm on Debian (sid), and I do have ogg libs (libogg-dev pkg) installed.

Any idea how to diagnose this?

Also, trying to bypass the ogg failure, over in the first tut, adding "mixerFlags=0" like this:

```
if game.init(GameWidth, GameHeight, title = GameTitle, integerScale = true, mixerFlags=0):
```

The first tut fails like this

CRITICAL: Can't initialize SDL_mixer: Failed loading udev_device_get_action: /usr/lib/x86_64-linux-gnu/libSDL2.so: undefined symbol: _udev_device_get_action

Main way I can get game.init to work, is to comment out the one section in nimgame.nim, like this:

```
  # # Init SDL_mixer
  # if mix.init(mixerFlags) == 0:
  #   sdl.logCritical(
  #     sdl.LogCategoryError, "Can't initialize SDL_mixer: %s", mix.getError())
  #   return false
```

**Comment by wizzardx** [comment]
Same error with OGG also happens for demo1, eg like this:


```
git clone https://github.com/Vladar4/nimgame2
cd nimgame2/demos/demo1
git checkout devel
nim c -r demo1.nim
```
However, things (demo1) seem to be fine if I remove ~/.nimble, change to that manual nimgame2 git checkout, and then run 'nimble install' from in there, followed by running 'nim c -r demo1'

tut 1 however at that point starts getting a new error (not seen in demo1), like this:

CRITICAL: Can't initialize SDL_mixer flags: Couldn't open /etc/timidity/freepats.cfg

I can get a bit further if I install freepats like this:

sudo apt-get install freepats

But then I start getting this error:

CRITICAL: Can't initialize SDL_mixer flags: Couldn't open /etc/timidity.cfg

Things go a bit further after this command:

sudo apt-get install timidity

But we still get this error when running tut 1:

CRITICAL: Can't initialize SDL_mixer flags: Couldn't open /usr/share/timidity/timidity.cfg

tl;dr, tut 1 seems entirely broken.

However demo1 seems to work, but only if you install nimgame2 from a local nimgame2 git clone (at least in devel branch, haven't tested in other branches yet).

I'm a bit confused here. Maybe something is a bit weird on my end; given that I'm using the 'debian sid' rolling release.

**Comment by wizzardx** [comment]
Update: I forgot to, in tut 1, take out my hack of "mixerFlags=0"

At that point tut 1 seems to work. Though the differing behavior of "mixerFlags=0" between the two versions of nimgame2 are confusing.

**Comment by wizzardx** [comment]
Things also seem to work fine in tut 1 if I install nimgame2 with nimble, like this:

nimble install nimgame2@#devel

It seems to just be the nimgame2@#head which has problems

**Comment by Vladar4** [comment]
It is a problem that was introduced with sdl_mixer v2.0.2. So you could either use sdl_mixer v2.0.1 with nimgame v0.5, or sdl_mixer v2.0.2 with devel branch.

Here's the commit 858beaed86c28f58c61ed68d7a6d6eb47c741135 that fixed it in devel.

Please tell if this helps.
